frontend/src/Components/BusinessHeader/BusinessHeader.jsx

```diff
@@ -10,17 +10,30 @@ const BusinessHeader = ({
   types,
 }) => {
   const [currentIndex, setCurrentIndex] = useState(0);
+  const [direction, setDirection] = useState(null);
 
   useEffect(() => {
     const interval = setInterval(() => {
-      setCurrentIndex((prevIndex) =>
-        prevIndex === photos.length - 1 ? 0 : prevIndex + 1
-      );
-    }, 2000); // Change image every 2 seconds
+      goToNext();
+    }, 4000); // Change image every 4 seconds
 
     return () => clearInterval(interval);
   }, [photos.length]);
 
+  const goToNext = () => {
+    setDirection("next");
+    setCurrentIndex((prevIndex) =>
+      prevIndex === photos.length - 1 ? 0 : prevIndex + 1
```

Comment:
Neat trick, totally optional, but I think you can reduce this to `(prevIndex + 1) % photos.length`